rubin_scheduler/scheduler/sim_runner.py

```diff
@@ -141,11 +141,14 @@ def sim_runner(
             # An observation failed to execute, usually it was outside
             # the altitude limits.
             if observatory.mjd == mjd_last_flush:
-                raise RuntimeError("Scheduler has failed to provide a valid observation multiple times.")
+                raise RuntimeError(
+                    "Scheduler has failed to provide a valid observation multiple times "
+                    f" at time ({observatory.mjd} from survey {scheduler.survey_index}."
+                )
             # if this is a first offence, might just be that targets set.
-            # Flush queue and get some new targets.
+            # Flush queue and try to get some new targets.
             scheduler.flush_queue()
-            mjd_last_flush = observatory.mjd + 0
+            mjd_last_flush = observatory.mjd
```

Comment:
Might need to be a `mjd_last_flush = copy.copy(observatory.mjd)` because that really needs to not be a pointer.